strongbox-data-service/src/main/java/org/carlspring/strongbox/gremlin/adapters/UnfoldEntityTraversal.java

```diff
@@ -0,0 +1,182 @@
+package org.carlspring.strongbox.gremlin.adapters;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.apache.tinkerpop.gremlin.process.traversal.Bytecode;
+import org.apache.tinkerpop.gremlin.process.traversal.Step;
+import org.apache.tinkerpop.gremlin.process.traversal.Traversal;
+import org.apache.tinkerpop.gremlin.process.traversal.TraversalSideEffects;
+import org.apache.tinkerpop.gremlin.process.traversal.TraversalStrategies;
+import org.apache.tinkerpop.gremlin.process.traversal.TraverserGenerator;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.process.traversal.step.TraversalParent;
+import org.apache.tinkerpop.gremlin.process.traversal.traverser.TraverserRequirement;
+import org.apache.tinkerpop.gremlin.structure.Graph;
+import org.carlspring.strongbox.data.domain.DomainObject;
+import org.carlspring.strongbox.gremlin.dsl.EntityTraversal;
+
+/**
+ * Traversal wrapper which is used to unfold entity on graph.
+ *
+ * @param <S>
+ * @param <E>
+ *
+ * @author sbespalov
+ */
+public class UnfoldEntityTraversal<S, E> implements EntityTraversal<S, E>
+{
+
+    private final String entityLabel;
+    private final DomainObject entity;
+    private final EntityTraversal<S, E> target;
+
+    public UnfoldEntityTraversal(String entityLabel,
+                                 DomainObject entity,
+                                 EntityTraversal<S, E> target)
+    {
+        this.entityLabel = entityLabel;
+        this.entity = entity;
+        this.target = target;
+    }
+
+    public String getEntityLabel()
+    {
+        return entityLabel;
+    }
+
+    public DomainObject getEntity()
+    {
+        return entity;
+    }
+
+    public EntityTraversal<S, E> getTarget()
+    {
+        return target;
+    }
+
+    @Override
+    public Bytecode getBytecode()
+    {
+        return target.getBytecode();
+    }
+
+    @Override
+    public List<Step> getSteps()
+    {
+        return target.getSteps();
+    }
```

Comment:
Just for my better understanding, what are `Step`-s?

Reply:
there is JavaDocs for `UnfoldEntityTraversal`, in general this is just delegate all to `org.apache.tinkerpop.gremlin.process.traversal.Traversal`